### PR TITLE
Damage roll selection in AI_CalcDamage

### PR DESCRIPTION
**Pull request by Pawkkie**
## Description
As is, `AI_CalcDamage` always assumes the lowest possible damage roll when calculating move damage. This PR adds a parameter to `AI_CalcDamage` that can specify using an alternate approach based on context, and adds an enum that includes highest, lowest, and average options to start with.

This PR also adds a very simple `AI_FLAG_CONSERVATIVE` to act as an inverse of `AI_FLAG_RISKY` that can be used to have trainers always use the lowest damage rolls when scoring their moves.

For all cases where this function is called, I've specified the rolls as follows. Happy to change any of these if we think it's a better idea:

- HasBadOdds: Assume player gets max roll (favour conservative switching decisions)
- Smart mon choices: Assume player gets max roll (favour conservative switching decisions)
- Best mon damage (a default case for vanilla switching): Assume AI gets an average roll
- Should use Z move: Assume AI gets an average roll
- Simulated damage: Use average rather than lowest by default for pre-calculating AI damage. Risky AI will instead use highest rolls, and conservative AI will instead use lowest rolls.

I should also note that the `HighestRollDmg` function currently accepts dmg and as input and immediately returns it as is. This is really just for future proofing, or very easily letting users change how they want bounds of the AI's damage rolls to behave without needing to change any of the other functions.

## **People who collaborated with me in this PR**
Just me writing it, but @iriv24 gave me the idea!

## **Discord contact info**
@pawkkie

### CODE REVIEW

**Review by DizzyEggg — DISMISSED** [review]
Imo it's a good idea, and I'm all for this change.

Why is the PR failing though?

**Comment by Pawkkie** [comment]
It's failing on exactly one test, which is one I wrote for the switch AI overhaul last fall, "AI_FLAG_SMART_MON_CHOICES: Mid-battle switches prioritize defensive options". In this test the AI should switch out their lead Ponyta to a defensive Aron that can tank the player's attacks. With this PR, the AI doesn't switch at all, and leaves Ponyta in to get KO'd.

I genuinely do not know why this test is failing. My project has deviated from expansion a bit and I'm pulling new stuff individually so it's hard to directly compare, but this is directly copied from the implementation I'm using which passes everything on my end.

I'll try to take some time to investigate further, but I'm very open to suggestions, especially regarding major AI stuff that's changed the last few months that might have introduced whatever AI logic changes have caused this deviation.

**Comment by Bassoonian** [comment]
Conflicts

**Comment by Pawkkie** [comment]
> Conflicts

Updated :)

File: src/battle_ai_util.c [hunk 1]
```diff
@@ -3304,8 +3326,8 @@ s32 AI_CalcPartyMonDamage(u32 move, u32 battlerAtk, u32 battlerDef, struct Battl
         SetBattlerData(battlerAtk); // set known opposing battler data
         AI_THINKING_STRUCT->saved[battlerAtk].saved = FALSE;
     }
-
-    dmg = AI_CalcDamage(move, battlerAtk, battlerDef, &effectiveness, FALSE, AI_GetWeather(AI_DATA));
+
```
Comment (Bassoonian) [inline src/battle_ai_util.c:3329]: Redundant spaces

**Comment by Pawkkie** [comment]
Spacing fixed :)

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**